processing/src/main/java/io/druid/query/filter/NotDimFilter.java

```diff
@@ -69,6 +79,24 @@ public Filter toFilter()
   }
 
   @Override
+  public RangeSet<String> getDimensionRangeSet(String dimension)
+  {
+    if (field instanceof AndDimFilter) {
```

Comment:
There are some potential cases involving null
* "NOT ( [0,INF) OR null)"
The inside of NOT would evaluate to null, and the complement would also be null. However, by breaking the NOT, this statement is "NOT([0,INF)) AND NOT(null)", which means it should actually evaluate to (-INF, 0).
* "NOT ( [0,INF) AND null )"
The inside of NOT would evaluate to [0,INF), and the complement would be (-INF, 0). However the statement is actually "NOT([0,INF)) OR NOT(null)", and it should be evaluated to null.

The special case for NOT is just to avoid stuff like "NOT(NOT(a) AND NOT(b)))" which would enter infinite loop when handling the AND.